src/transactions/test/TxEnvelopeTests.cpp

```diff
@@ -1532,8 +1533,11 @@ TEST_CASE("txenvelope", "[tx][envelope]")
             StellarValue sv = app->getHerder().makeStellarValue(
                 txSet->getContentsHash(), 1, emptyUpgradeSteps,
                 app->getConfig().NODE_SEED);
-            LedgerCloseData ledgerData(1, txSet, sv);
-            app->getLedgerManager().closeLedger(ledgerData);
+            auto& herder = static_cast<HerderImpl&>(app->getHerder());
+            herder.getPendingEnvelopes().putTxSet(txSet->getContentsHash(), 2,
```

Comment:
should this be a test only method exposed by `Herder` (this avoids importing `HerderImpl` and duplicate code) -- this is similar to the 3 param `recvSCPEnvelope(SCPEnvelope const& envelope, const SCPQuorumSet& qset, TxSetFrame txset)`

Reply:
sure, I added a new test method to externalize a value.